lib/iris/fileformats/netcdf.py

```diff
@@ -789,6 +802,10 @@ def load_cubes(filenames, callback=None):
         for cf_var in data_variables:
             cube = _load_cube(engine, cf, cf_var, filename)
 
+            # Call a post-processing completion step for each cube, if provided.
```

Comment:
This approach is based on a "smallest necessary changes" strategy,
which was appropriate the original "ugrid proof-of-concept" work phase, but is perhaps a bit untidy now.

The original code views the 'cf' object (actually, a CFReader) as a box of information to draw on, and the _load_cube call is in charge of the cube.
The _load_cube call "knows" about the pyke engin;  the cf data structures in the reader; and how to construct a cube from that information.
This new callback hook approach is now giving the reader a degree of "final control" over each cube, which was not there before.

Reply:
As discussed offline - happy with this pragmatic approach for now given what we're working with as a starting point. Could be refactored further at a future date if/when Pyke rules are stripped out.